lib/signaling/v2/peerconnection.js

```diff
@@ -488,10 +502,46 @@ class PeerConnectionV2 extends StateMachine {
       this._log.info('ICE reconnected');
     }
 
+    if (iceConnectionState === 'connected') {
+      this._mediaMonitor.start(() => {
```

Comment:
@makarandp0 This can be:
```js
this._iceConnectionMonitor.checkIfInactive().then(() => {
  if (this._lastIceConnectionState !== 'disconnected') {
    return this._iceConnectionMonitor.checkIfInactive();
  }
  if (!this._shouldRestartIce && !this._isRestartingIce) {
    this._shouldRestartIce = true;
    this.offer();
  }
});
```

Reply:
ah, Now I understand what you mean by using promise for this api -  Its using nested promises -  I think there are couple cons with this approach.

1. nested promises are making code bit complicated.
2. this usage makes it difficult to use "previous" state from the media monitor. For example handling of this case is not obvious.
a) connected: time 0 -  bytes = 0 - Api was called and it returned a promise..
b) connected: time 4 -  bytes = 0 - promise resolved. 
c) Api was was called again   because state is still connected.
d) disconnected:
e) disconnected: time 5 - bytes = 0.

At e) should the API resolve the promise? For callback it's obvious that it should fire again (as it was started, and callbacks on every inactivity). But since a new promise was started c) and it hasn't yet been 3 seconds of inactivity (although total inactivity is 5 seconds).


3. Also  with this promise API the promise never gets cleared (as there is no `stop` functionality) That means we will always have an outstanding promise (and associated timer) for peer connection - long after the peer connection is closed. This would lead to memory leaks, if there were lots of peerconnection on the app.